nilearn/datasets/func.py

```diff
@@ -1669,3 +1669,219 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
 
     return Bunch(func=func, mat_files=mat, phenotypic=csv_array,
                  description=fdescr)
+
+
+def fetch_surf_nki_enhanced(n_subjects=30, data_dir=None,
+                            url=None, resume=True, verbose=1):
+    """Download and load the NKI enhanced resting-state dataset,
+       preprocessed and projected to the surface.
+
+    Parameters
+    ----------
+    n_subjects: int, optional
+        The number of subjects to load from maximum of 102 subjects.
+        By default, 30 subjects will be loaded. If None is given,
+        all 102 subjects will be loaded.
+
+    data_dir: string, optional
+        Path of the data directory. Used to force data storage in a specified
+        location. Default: None
+
+    url: string, optional
+        Override download URL. Used for test only (or if you setup a mirror of
+        the data). Default: None
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, the interest attributes are :
+         - 'func': Paths to functional resting-state images
```

Comment:
Also, worth to write that the images we return are Gifti-like images.

I would write
```python 
- 'func_right' : Gifti-like images
      Paths to functional resting-state images
```